src/layers/mpls.rs

```diff
@@ -0,0 +1,151 @@
+    exp: u8, //This is only 3 bits
+    // #[serde(serialize_with = "crate::types::hex::serialize_lower_hex_u8")]
+    bos: bool, //This is only 1 bit
+    #[serde(serialize_with = "crate::types::hex::serialize_lower_hex_u8")]
+    ttl: u8, //This is 1 byte
+}
+
+#[derive(Debug, Default, Serialize, Clone)]
+pub struct MPLS {
+    labels: Vec<MPLSLabel>,
+}
+
+impl MPLS {
+    pub fn creator() -> Box<dyn Layer + Send> {
+        Box::<MPLS>::default()
+    }
+}
+
+impl Layer for MPLS {
+    fn decode_bytes(
+        &mut self,
+        bytes: &[u8],
+    ) -> Result<(Option<Box<dyn Layer + Send>>, usize), Error> {
+        if bytes.len() < MPLS_HDR_LEN {
+            return Err(Error::TooShort {
+                required: MPLS_HDR_LEN,
+                available: bytes.len(),
+                data: hex::encode(bytes),
+            });
+        }
+
+        let mut label_index = 0;
+
+        loop {
+            let mut current_label = MPLSLabel {
+                label: 0,
+                exp: 0,
+                bos: false,
+                ttl: 0,
+            };
+            let byte_offset = 4 * label_index;
+
+            //FIXME:For now the first few bits are not useful in label,exp and bos
+            current_label.label = u32::from_be_bytes(
+                bytes[(byte_offset + 0)..(byte_offset + 4)]
+                    .try_into()
+                    .unwrap(),
+            ) as u32
+                >> 12;
+            current_label.exp = u8::from_be(bytes[byte_offset + 2] << 4) >> 5;
+            current_label.bos = bytes[byte_offset + 2] & 0x01 == 0x01;
+            current_label.ttl = bytes[byte_offset + 3];
+
+            self.labels.push(current_label);
+            label_index += 1;
+
+            if current_label.bos == true {
+                break;
+            }
+        }
+
+        //FIXME:Support IPV6
+        Ok((Some(IPv4::creator()), 4 * (label_index)))
+    }
+
+    fn name(&self) -> &'static str {
+        "MPLS"
+    }
+
+    fn short_name(&self) -> &'static str {
+        "mpls"
+    }
+}
+
+#[cfg(test)]
+mod tests {
+
+    use crate::layers;
+    use crate::{Layer, Packet, ENCAP_TYPE_ETH};
+
+    #[test]
+    fn parse_valid_mpls_packet() {
+        let _ = layers::register_defaults();
+
+        let mpls_packet = vec![
+            0x00, 0x30, 0x96, 0xe6, 0xfc, 0x39, 0x00, 0x30, 0x96, 0x05, 0x28, 0x38, 0x88, 0x47,
+            0x00, 0x01, 0xdd, 0xff, 0x45, 0xc0, 0x00, 0x28, 0x00, 0x03, 0x00, 0x00, 0xff, 0x06,
+            0xa4, 0xe8, 0x0a, 0x01, 0x02, 0x01, 0x0a, 0x22, 0x00, 0x01, 0x2a, 0xf9, 0x00, 0x17,
+            0x98, 0x32, 0x10, 0x05, 0x8d, 0xd5, 0x8e, 0xa5, 0x50, 0x10, 0x10, 0x20, 0x99, 0xcd,
+            0x00, 0x00, 0x00, 0x00,
+        ];
+
+        let mut mpls: Box<dyn Layer> = Box::new(super::MPLS::default());
+
+        let p = mpls.decode_bytes(&mpls_packet[28..]);
+        assert!(p.is_ok(), "{:#?}", mpls);
+    }
+
+    #[test]
```

Comment:
If this test is failing - you can `ignore` this test during the CI run. If this is the only one that is failing, we can then fix the issue later, by creating an issue. For now simply adding `#[ignore]` tag and then writing a `// TODO: ` is good enough.

Reply:
The tests are passing successfully, since it's the same packet I'd used in the example. So I believe we don't need to make any changes.